File: business/src/main/java/work/beltran/discogsbrowser/business/CollectionInteractor.java
package work.beltran.discogsbrowser.business;

import io.reactivex.Completable;
import io.reactivex.Single;
import work.beltran.discogsbrowser.api.model.UserCollection;

/**
 * Created by Miquel Beltran on 8/27/16
 * More on http://beltran.work
 */
public interface CollectionInteractor {
    Single<UserCollection> getCollection(int page);
    Single<UserCollection> getRecord(int recordId);
    Completable addRecord(int recordId);
    Completable removeRecord(int recordId);
}

File: business/src/main/java/work/beltran/discogsbrowser/business/CollectionInteractor.kt
package work.beltran.discogsbrowser.business

import io.reactivex.Completable
import io.reactivex.Single
import work.beltran.discogsbrowser.api.model.UserCollection

/**
 * Created by Miquel Beltran on 8/27/16
 * More on http://beltran.work
 */
interface CollectionInteractor {
    fun getCollection(page: Int): Single<UserCollection>
    fun getRecord(recordId: Int): Single<UserCollection>
    fun addRecord(recordId: Int): Completable
    fun removeRecord(recordId: Int): Completable
}
